cirq-google/cirq_google/engine/abstract_engine.py

```diff
@@ -134,3 +139,35 @@ def get_sampler(self, processor_id: str | list[str]) -> cirq.Sampler:
             processor_id: String identifier, or list of string identifiers,
                 determining which processors may be used when sampling.
         """
+
+    @abc.abstractmethod
+    def get_processor_config_by_snapshot_id(
```

Comment:
The naming of these methods seems to be a bit inconsistent across the different files. Let's standardize to `get_processor_config_from_snapshot` and `get_processor_config_from_run`. The "processor" bit can be omitted from the `AbstractProcessor` and its subclasses.